### PR TITLE
Feature/489

### LINKED ISSUE #489: Add more tests for Ubiquity Dollar custom smart contracts
### Notice: tests must achieve 80% or greater code coverage per smart contract.

# Add more tests for the following smart contracts

- `src/ubiquistick/UbiquiStick.sol`
- `src/ubiquistick/UbiquiStickSale.sol`

| File                                               | % Lines           | % Statements      | % Branches        | % Funcs          |
|----------------------------------------------------|-------------------|-------------------|-------------------|------------------|
| src/ubiquistick/UbiquiStick.sol                    | 42.31% (11/26)    | 39.29% (11/28)    | 0.00% (0/12)      | 57.14% (8/14)    |
| src/ubiquistick/UbiquiStickSale.sol                | 0.00% (0/13)      | 0.00% (0/15)      | 0.00% (0/6)       | 0.00% (0/6)      |

- `src/ubiquistick/LP.sol`
- `src/ubiquistick/SimpleBond.sol`
- `src/ubiquistick/UAR.sol`

| File                                               | % Lines           | % Statements      | % Branches        | % Funcs          |
|----------------------------------------------------|-------------------|-------------------|-------------------|------------------|
| src/ubiquistick/LP.sol                             | 0.00% (0/1)       | 0.00% (0/1)       | 100.00% (0/0)     | 0.00% (0/1)      |
| src/ubiquistick/SimpleBond.sol                     | 84.91% (45/53)    | 86.44% (51/59)    | 44.44% (8/18)     | 85.71% (12/14)   |
| src/ubiquistick/UAR.sol                            | 0.00% (0/1)       | 0.00% (0/1)       | 100.00% (0/0)     | 0.00% (0/1)      |

- `src/dollar/UbiquityChef.sol`
- `src/dollar/UbiquityFormulas.sol`

| File                                               | % Lines           | % Statements      | % Branches        | % Funcs          |
|----------------------------------------------------|-------------------|-------------------|-------------------|------------------|
| src/dollar/UbiquityChef.sol                        | 88.41% (61/69)    | 88.16% (67/76)    | 61.11% (11/18)    | 100.00% (15/15)  |
| src/dollar/UbiquityFormulas.sol                    | 40.74% (11/27)    | 42.55% (20/47)    | 0.00% (0/2)       | 40.00% (2/5)     |

- `src/dollar/core/CreditRedemptionCalculator.sol`
- `src/dollar/core/DollarMintExcess.sol`
- `src/dollar/core/UbiquityCreditToken.sol`

| File                                               | % Lines           | % Statements      | % Branches        | % Funcs          |
|----------------------------------------------------|-------------------|-------------------|-------------------|------------------|
| src/dollar/core/CreditRedemptionCalculator.sol     | 0.00% (0/9)       | 0.00% (0/14)      | 0.00% (0/2)       | 0.00% (0/3)      |
| src/dollar/core/DollarMintExcess.sol               | 0.00% (0/30)      | 0.00% (0/37)      | 0.00% (0/2)       | 0.00% (0/4)      |
| src/dollar/core/UbiquityCreditToken.sol            | 0.00% (0/2)       | 0.00% (0/3)       | 100.00% (0/0)     | 0.00% (0/1)      |

### PR DESCRIPTION
**Pull request by rndquu**
Resolves #489 

1. Removed [UAR.sol](https://github.com/ubiquity/ubiquity-dollar/blob/development/packages/contracts/src/ubiquistick/UAR.sol) because it is not being used anywhere
2. Updated coverage:

| File                                               | % Lines           | % Statements      | % Branches        | % Funcs          |
|--------------------------------|-------------------|-------------------|-------------------|------------------|
| src/ubiquistick/UbiquiStick.sol                    | 100.00% (26/26)   | 100.00% (28/28)   | 100.00% (12/12)   | 100.00% (14/14)  |
| src/ubiquistick/UbiquiStickSale.sol                | 100.00% (13/13)   | 100.00% (15/15)   | 100.00% (6/6)     | 100.00% (6/6)    |
| src/ubiquistick/LP.sol                             | 100.00% (1/1)     | 100.00% (1/1)     | 100.00% (0/0)     | 100.00% (1/1)    |
| src/ubiquistick/SimpleBond.sol                     | 85.19% (46/54)    | 86.44% (51/59)    | 44.44% (8/18)     | 85.71% (12/14)   |
| src/dollar/UbiquityChef.sol                        | 88.41% (61/69)    | 88.16% (67/76)    | 61.11% (11/18)    | 100.00% (15/15)  |
| src/dollar/UbiquityFormulas.sol                    | 100.00% (27/27)   | 100.00% (47/47)   | 100.00% (2/2)     | 100.00% (5/5)    |
| src/dollar/core/CreditRedemptionCalculator.sol     | 100.00% (9/9)     | 100.00% (14/14)   | 100.00% (2/2)     | 100.00% (3/3)   |
| src/dollar/core/DollarMintExcess.sol               | 100.00% (27/27)   | 100.00% (37/37)   | 50.00% (1/2)      | 100.00% (4/4)    |
| src/dollar/core/UbiquityCreditToken.sol            | 100.00% (2/2)     | 100.00% (3/3)     | 100.00% (0/0)     | 100.00% (1/1)    |

### CODE REVIEW

File: packages/contracts/test/dollar/core/CreditRedemptionCalculator.t.sol [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            abi.encodeWithSelector(IERC20.totalSupply.selector),
+            abi.encode(1)
+        );
+        vm.expectRevert("Credit to Dollar: DEBT_TOO_HIGH");
+        creditRedemptionCalculator.getCreditAmount(1 ether, 10);
+    }
+
+    function testGetCreditAmount_ShouldReturnAmount() public {
+        uint amount = creditRedemptionCalculator.getCreditAmount(1 ether, 10);
+        assertEq(amount, 9999999999999999999);
```
Comment (0x4007) [inline packages/contracts/test/dollar/core/CreditRedemptionCalculator.t.sol]: Whats the `9999999999999999999` about?

File: packages/contracts/test/dollar/UbiquityFormulas.t.sol [hunk 2]
```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
```
Comment (0x4007) [inline packages/contracts/test/dollar/UbiquityFormulas.t.sol]: Shouldn't we be using the same pragma everywhere? 0.8.16?

File: packages/contracts/test/ubiquistick/LP.t.sol [hunk 3]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import "forge-std/Test.sol";
+
+import "../../src/ubiquistick/LP.sol";
```
Comment (0x4007) [inline packages/contracts/test/ubiquistick/LP.t.sol]: I'm actually not sure the purpose of this file and deleted it on my refactoring branch.

File: packages/contracts/test/dollar/core/CreditRedemptionCalculator.t.sol (hunk 1 above)
Comment (rndquu) [reply packages/contracts/test/dollar/core/CreditRedemptionCalculator.t.sol]: > Whats the `9999999999999999999` about?

the amount in wei of credit tokens which user gets for burning 1 dollar at block №10

Comment (rndquu) [reply packages/contracts/test/dollar/core/CreditRedemptionCalculator.t.sol]: I have a plan of writing some intergration tests to cover basic scenarios (like TWAP > 1$, TWAP < 1$, etc...) + update docs with basic examples. This is where we will make sure that `CreditRedemptionCalculator` works as expected.

File: packages/contracts/test/dollar/UbiquityFormulas.t.sol (hunk 2 above)
Comment (0x4007) [reply packages/contracts/test/dollar/UbiquityFormulas.t.sol]: @rndquu

**Comment by 0x4007** [comment]
@rndquu please resolve all the pending conversations before we merge.

**Comment by rndquu** [comment]
> @rndquu please resolve all the pending conversations before we merge.

Yes, after sync with the `development` branch some tests are broken so I need time to fix them

File: packages/contracts/test/ubiquistick/LP.t.sol (hunk 3 above)
Comment (rndquu) [reply packages/contracts/test/ubiquistick/LP.t.sol]: removed `LP.t.sol`

File: packages/contracts/test/dollar/UbiquityFormulas.t.sol (hunk 2 above)
Comment (rndquu) [reply packages/contracts/test/dollar/UbiquityFormulas.t.sol]: updated

**Review by 0x4007 — APPROVED** [review]

**Comment by 0x4007** [comment]
@0xcodercrane

**Review by cuddle-dev — APPROVED** [review]

**Pull request merged**